backend/grants/admin.py

```diff
@@ -548,88 +552,137 @@ def save_form(self, request, form, change):
-
-        results = []
-        for country_code, group in groupby(list(qs), key=lambda k: k.travelling_from):
-            country = countries.get(code=country_code)
-            if not country:
-                continue
-            grants = list(group)
-            counter = Counter([g.status for g in grants])
-            results.append(
-                {
-                    "continent": country.continent.name,
-                    "country": country.name,
-                    "total": int(sum([g.total_amount for g in grants])),
-                    "count": len(grants),
-                    "rejected": counter.get(Grant.Status.rejected, 0),
-                    "approved": counter.get(Grant.Status.approved, 0),
-                    "waiting_list": counter.get(Grant.Status.waiting_list, 0)
-                    + counter.get(Grant.Status.waiting_list_maybe, 0),
-                    "waiting_for_confirmation": counter.get(
-                        Grant.Status.waiting_for_confirmation, 0
-                    ),
-                    "refused": counter.get(Grant.Status.refused, 0),
-                    "confirmed": counter.get(Grant.Status.confirmed, 0),
-                }
-            )
-
-        results = sorted(results, key=lambda k: (k["continent"], k["country"]))
-
-        counter = Counter([g.status for g in qs])
-        footer = {
-            "title": "Total",
-            "count": len(qs),
-            "total": int(
-                sum(
-                    [
-                        g.total_amount
-                        for g in qs
-                        if g.status
-                        in (
-                            Grant.Status.confirmed,
-                            Grant.Status.approved,
-                            Grant.Status.waiting_for_confirmation,
-                        )
-                    ]
-                )
-            ),
-            "rejected": counter.get(Grant.Status.rejected, 0),
-            "approved": counter.get(Grant.Status.approved, 0),
-            "waiting_list": counter.get(Grant.Status.waiting_list, 0)
-            + counter.get(Grant.Status.waiting_list_maybe, 0),
-            "waiting_for_confirmation": counter.get(
-                Grant.Status.waiting_for_confirmation, 0
+    def get_urls(self):
+        urls = super().get_urls()
+        custom_urls = [
+            path(
+                "summary/",
+                self.admin_site.admin_view(self.summary_view),
+                name="grants-summary",
             ),
-            "refused": counter.get(Grant.Status.refused, 0),
-            "confirmed": counter.get(Grant.Status.confirmed, 0),
+        ]
+        return custom_urls + urls
+
+    def summary_view(self, request):
+        """
+        Custom view for summarizing Grant data in the Django admin.
+        Aggregates data by country and status, and applies request filters.
+        """
+        # Initialize statuses
```

Comment:
I don't think we need those comments? "Initialize statuses" "Apply filters to the Grant queryset" etc